src/storage/src/sink/kafka.rs

```diff
@@ -937,21 +896,16 @@ impl KafkaSinkState {
             if min_frontier > self.latest_progress_ts {
                 // record the write frontier in the progress topic.
                 if let Some(progress_state) = self.sink_state.unwrap_running() {
-                    if self.transactional {
-                        self.retry_on_txn_error(|p| p.begin_transaction()).await?;
-                    }
+                    self.retry_on_txn_error(|p| p.begin_transaction()).await;
```

Comment:
same for this function (which is perhaps complicated if we choose to return an error from `retry_on_txn_error` instead of panicking).  I think this is a spot where the `let () = ...` makes it extra clear to the reader that we're not accidentally throwing away an error.

The function has quite a few calls that I think would benefit in the same way